emukit/examples/profet/performance_assessment.py

```diff
@@ -0,0 +1,78 @@
+import numpy as np
+import scipy.stats as stats
+
+
+def compute_runtime_feval(trajectory: np.ndarray, target: float) -> int:
+    """
+    Computes the runtime in terms of number of function evaluations until a performance smaller or
+    equal than a specified target value is achieved.
+
+    :param trajectory: trajectory of the performance of an optimizer after each function evaluation
+    :param target: target value
+    :return: runtime
+    """
+    rt = np.where(trajectory <= target)[0]
+    if len(rt) == 0:
+        rt = np.inf
+        return rt
+    else:
+        return rt[0] + 1
+
+
+def compute_ecdf(error: np.ndarray, targets: np.ndarray) -> dict:
+    """
+    Computes the empirical cumulative distribution (ECDF) of the runtime of an optimizer
+    across different targets and tasks.
+
+    :param error: matrix with I x R x N entries, where I are the number of instances or tasks,
+    R is the number of runs per task and N are the number of function evaluations per task and run
+    :param targets: matrix with I x T entries, where I are the number of instances or tasks and T are the number
+    of targets values
+    :return: ECDF as dict, where 'x' defines the runtime and 'y' the CDF
+    """
+    n_instances = error.shape[0]
+    n_runs = error.shape[1]
+    assert targets.shape[0] == n_instances
```

Comment:
Let's avoid asserts. If this code can go wrong, let's throw an exception with message, giving end user some information

Reply:
ok done